File: java/pattern/src/factory/abstractfactory/Seat.java
package factory.abstractfactory;

public interface Seat {
    void massage();
}

class LuxurySeat implements Seat {

    @Override
    public void massage() {
        System.out.println("可以按摩");
    }
}

class LowSeat implements Seat {

    @Override
    public void massage() {
        System.out.println("硬座");
    }
}

File: php/factory/abstractfactory/Seat.php
<?php

/**
 * 汽车座椅
 */

interface Seat
{
    public function massage();
}

/**
 * 高级车的座椅
 */
class LuxurySeat implements Seat {
    public function massage()
    {
        echo '可以按摩';
    }
}


class LowSeat implements Seat {
    public function massage()
    {
        echo '硬座';
    }
}


